package/libsolv.spec.in

```diff
@@ -190,83 +165,80 @@ Group:          Development/Languages/Python
 Python3 bindings for libsolv.
 
 %package -n perl-solv
-Requires:       perl = %{perl_version}
 Summary:        Perl bindings for the libsolv library
 Group:          Development/Languages/Perl
+Requires:       perl = %{perl_version}
 
 %description -n perl-solv
 Perl bindings for libsolv.
 
 %prep
-%setup -n libsolv-%{version}
+%setup -q
 
 %build
 export CFLAGS="%{optflags}"
 export CXXFLAGS="$CFLAGS"
 
 CMAKE_FLAGS=
-%if 0%{?fedora} || 0%{?rhel} >= 6
-CMAKE_FLAGS="-DFEDORA=1 -DENABLE_APPDATA=1 -DENABLE_COMPS=1"
+%if 0%{?fedora} || 0%{?rhel}
+CMAKE_FLAGS="-DFEDORA=1"
 %endif
 %if 0%{?mageia}
-CMAKE_FLAGS="-DMAGEIA=1 -DENABLE_APPDATA=1 -DENABLE_COMPS=1"
+CMAKE_FLAGS="-DMAGEIA=1"
 %endif
 %if 0%{?suse_version}
-CMAKE_FLAGS="-DSUSE=1 -DENABLE_APPDATA=1 -DENABLE_COMPS=1"
+CMAKE_FLAGS="-DSUSE=1"
 %endif
 
-cmake	$CMAKE_FLAGS \
-	-DCMAKE_INSTALL_PREFIX=%{_prefix} \
-	-DLIB=%{_lib} \
-	-DCMAKE_VERBOSE_MAKEFILE=TRUE \
-	-DCMAKE_BUILD_TYPE=RelWithDebInfo \
+# SHARE/MODULE flags in SUSE enforce no-undefined which is not applicable for
+# bindings here
+%cmake $CMAKE_FLAGS \
+	-DCMAKE_MODULE_LINKER_FLAGS="-Wl,--as-needed -Wl,-z,now" \
+	-DCMAKE_SHARED_LINKER_FLAGS="-Wl,--as-needed -Wl,-z,now" \
 	-DWITH_LIBXML2=1 \
+	-DENABLE_APPDATA=1 \
+	-DENABLE_COMPS=1 \
 	%{?with_enable_static:-DENABLE_STATIC=1} \
 	%{?with_disable_shared:-DDISABLE_SHARED=1} \
-	%{?with_perl_binding:-DENABLE_PERL=1} \
-	%{?with_python_binding:-DENABLE_PYTHON=1} \
-	%{?with_python3_binding:-DENABLE_PYTHON3=1} \
-	%{?with_ruby_binding:-DENABLE_RUBY=1} \
+	%{?with_perl:-DENABLE_PERL=1} \
+	%{?with_python:-DENABLE_PYTHON=1} \
+	%{?with_python3:-DENABLE_PYTHON3=1} \
+	%{?with_ruby:-DENABLE_RUBY=1} \
 	%{?with_bz2:-DENABLE_BZIP2_COMPRESSION=1} \
 	%{?with_xz:-DENABLE_LZMA_COMPRESSION=1} \
 	%{?with_richdeps:-DENABLE_COMPLEX_DEPS=1} \
 	%{?with_zypp:-DENABLE_SUSEREPO=1 -DENABLE_HELIXREPO=1} \
-	-DUSE_VENDORDIRS=1 \
-	-DCMAKE_SKIP_RPATH=1
+	-DUSE_VENDORDIRS=1
 make %{?_smp_mflags}
 
 %install
-make DESTDIR=%{buildroot} install
-%if 0%{?suse_version}
-%if %{with python_binding}
-pushd %{buildroot}/%{python_sitearch}
-python %py_libdir/py_compile.py *.py
-python -O %py_libdir/py_compile.py *.py
-popd
+%cmake_install
+
+%if %{with python2}
+%py_compile %{buildroot}/%{python_sitearch}
 %endif
-%if %{with python3_binding}
+%if %{with python3}
 %py3_compile %{buildroot}/%{python3_sitearch}
 %endif
-%endif
 %if %{with disable_shared}
 # we want to leave the .a file untouched
 export NO_BRP_STRIP_DEBUG=true
 %endif
 
-%if !%{with disable_shared}
-%post -n libsolv@LIBSOLV_SOVERSION@ -p /sbin/ldconfig
+%check
+%ctest
 
-%postun -n libsolv@LIBSOLV_SOVERSION@ -p /sbin/ldconfig
+%if !%{with disable_shared}
+%post -n %{libname} -p /sbin/ldconfig
+%postun -n %{libname} -p /sbin/ldconfig
 
-%files -n libsolv@LIBSOLV_SOVERSION@
-%defattr(-,root,root)
+%files -n %{libname}
 %doc LICENSE*
```

Comment:
Change this to a proper `%license` usage with backward compat.
```
%{!?_licensedir:%global license %doc}
%license LICENSE*
```

Also, because shared isn't built by default, this needs to be present in *every* libsolv subpackage.

Reply:
Why, again SUSE is the target distro, rest is nice benefit.

Reply:
That doesn't change that the license isn't included in all the subpackages that get installed. Since you don't use the shared library, this is necessary to ensure the license is shipped.